lxd/endpoints/network_test.go

```diff
@@ -88,3 +90,22 @@ func newTCPListener(t *testing.T) *net.TCPListener {
 	require.NoError(t, err)
 	return listener
 }
+
+// Create IPv4 0.0.0.0 listener using random port
+// and ensure it is not accessible via IPv6 request.
+func TestEndpoints_NetworkCreateTCPSocketIPv4(t *testing.T) {
+	endpoints, config, cleanup := newEndpoints(t)
+	defer cleanup()
+
+	config.NetworkAddress = "0.0.0.0:0"
+	require.NoError(t, endpoints.Up(config))
+
+	address, certificate := endpoints.NetworkAddressAndCert()
+	parts := strings.Split(address, ":")
+	ipv6Address := fmt.Sprintf("[::1]:%s", parts[1])
+
+	assert.NoError(t, httpGetOverTLSSocket(address, certificate))
```

Comment:
Can this use 127.0.0.1 rather than address from listener to show sane behavior as ipv6 request?

Reply:
So I mean use a var called ipv4Address like the ipv6 approach.